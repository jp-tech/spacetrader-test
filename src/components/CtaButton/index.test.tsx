import { it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import Button from "./";

const testText = "Press Me!"

it('renders correctly', async () => {
    const onChange = vi.fn();
    const {container, getByRole} = render(
        <Button
            onClick={onChange}
        >
            {testText}
        </Button>
    );
    getByRole('button').click();
    expect(container).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledOnce();
});
