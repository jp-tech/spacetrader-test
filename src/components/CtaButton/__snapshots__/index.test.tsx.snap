// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-5r9pwp-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="submit"
  >
    Press Me!
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
